cli/cmd/completion.go

```diff
@@ -46,17 +48,54 @@ func RootShellCompletionCommands(cmd *cobra.Command, args []string,
 	return commands, cobra.ShellCompDirectiveDefault
 }
 
+// injectRocksCompletion combines luarocks completions with cobra completions.
+func injectRocksCompletion(shell string, completion []byte) ([]byte, error) {
+	label := []byte(`    # The user could have moved the cursor backwards on the command-line.`)
+
+	injection, err := util.ReadEmbedFile(rocks.EmbedCompletions, "completions/"+shell+"_injection")
+	if err != nil {
+		return nil, err
+	}
+	rocks, err := util.ReadEmbedFile(rocks.EmbedCompletions, "completions/"+shell+"_rocks")
+	if err != nil {
+		return nil, err
+	}
+
+	res := bytes.Buffer{}
+	idx := bytes.Index(completion, label)
+	if idx == -1 {
+		return nil, fmt.Errorf("failed to inject LuaRocks completions")
+	}
+	res.Write(completion[:idx])
+	res.Write([]byte(injection))
+	res.Write(completion[idx:])
+	res.Write([]byte(rocks))
+
+	return res.Bytes(), nil
```

Comment:
Let's add a test to check that at least `source` command works fine after this injection.

Reply:
Test added.